packages/contracts/package.json

```diff
@@ -8,15 +8,15 @@
     "chain": "yarn rm-dbdata && anvil --config-out localhost.json",
     "clean": "forge clean && rm -rf broadcast cache deployments types localhost.json index.js",
     "dev": "yarn rm-dbdata && anvil --config-out localhost.json & make deploy-local",
-    "build": "forge build && yarn gen-types",
+    "build": "forge build && yarn gen-types && tsc",
     "test": "forge test",
     "format": "forge fmt && prettier './script/**/*.js' --write",
     "lint": "forge fmt && eslint --fix --ignore-path .gitignore && yarn solhint './*(test|src)/**/*.sol'",
     "db-check": "tsc && node ./script/utils/checkDb.js",
     "db-export": "tsc && node ./script/utils/exporter.js",
     "deploy": "make deploy-local",
-    "deploy:ci": "yarn db-export && make deploy-baseSepolia",
-    "gen-types": "yarn typechain --target ethers-v5 --out-dir ./types ./out/*.sol/*.json && tsc",
```

Comment:
gen-types should come after deploy, shouldn't it?  if we remove the tsc here, we won't compile the types until another build call.

Reply:
`gen-types` should come after a `forge build`. And yes, a `tsc` compiles all Typescript. However, having `tsc` in a 'generate code' command is not 'clean', to my mind. Convenient, yes, but not clean and not really necessary, as long as the TS compilation/transpilation always happens.
So, I changed the `build` script to include all 3 build-like tasks (compile, generate, transpile).